auctions/dutch-auction/src/interface.sw

```diff
@@ -0,0 +1,164 @@
+library interface;
+
+dep data_structures;
+
+use data_structures::Auction;
+
+abi DutchAuction {
+    /// Returns the number of active auctions of the author
+    ///
+    /// # Argumets
+    /// 
+    /// * `author` - The author of which to retrieve the number of active auctions of
+    #[storage(read)]
+    fn active_auctions_of_author(author: Identity) -> u64;
+    /// Returns the auction_id of the active auction of the author at the given index
+    ///
+    /// # Arguments
+    /// 
+    /// * `author` - The author of which to retrieve the active auction of
+    /// * `index` - The index to retrieve
+    ///
+    /// # Reverts
+    ///
+    /// * When the index is greater than the number of active auctions of the author
+    #[storage(read)]
+    fn active_auction_of_author(author: Identity, index: u64) -> u64;
+    
+    /// Returns the auction data for the specified auction ID
+    ///
+    /// # Arguments
+    ///
+    /// * `auction_id` - The id of the auction of which to fetch the data of 
+    /// 
+    /// # Reverts
+    ///
+    /// * When the auction_id is 0 or greater than storage.auction_count
+    #[storage(read)]
+    fn auction(auction_id: u64) -> Auction;
+    /// Returns the number of auctions created by author
+    ///
+    /// # Arguments
+    ///
+    /// * `author` - The Identity of which to check the number of auctions created
+    #[storage(read)]
+    fn auctions_of_author(author: Identity) -> u64;
+    /// Returns the number of auctions some bidder has won
+    ///
+    /// # Arguments
+    ///
+    /// * `bidder` - The Identity of which to check the number of auctions won
+    #[storage(read)]
+    fn auctions_won(bidder: Identity) -> u64;
+    /// Returns the auction_id of the auction of the author at the given index
+    ///
+    /// # Arguments
+    /// 
+    /// * `author` - The author of which to retrieve the auction of
+    /// * `index` - The index to retrieve
+    ///
+    /// # Reverts
+    ///
+    /// * When the index is greater than the number of auctions of the author
+    #[storage(read)]
+    fn auction_of_author(author: Identity, index: u64) -> u64;
```

Comment:
This is too confusing. There are too many functions for `active` vs `inactive`. Just change the functions to have 1 source of truth which indicates if something is active or not rather than have 2 functions for each thing. I'll review the remaining docs after that is done.